examples/6field-simple/elm_6f.cxx

```diff
@@ -12,1677 +12,1682 @@
-Field3D vth_i;              // Ion Thermal Velocity profile (M / S)
-Field3D vth_e;              // Electron Thermal Velocity profile (M / S)
-Field3D kappa_par_i;        // Ion Thermal Conductivity profile (kg&M / S^2)
-Field3D kappa_par_e;        // Electron Thermal Conductivity profile (kg*M / S^2)
-Field2D omega_ci, omega_ce; // cyclotron frequency
-Field3D kappa_perp_i;       // Ion perpendicular Thermal Conductivity profile (kg&M / S^2)
-Field3D kappa_perp_e; // Electron perpendicular Thermal Conductivity profile (kg*M / S^2)
-
-bool output_transfer; // output the results of energy transfer
-bool output_ohm;      // output the results of the terms in Ohm's law
-bool output_flux_par; // output the results of parallel particle and heat flux
-Field3D T_M, T_R, T_ID, T_C,
-    T_G; // Maxwell stress, Reynolds stress, ion diamagbetic and curvature term
-Field3D ohm_phi, ohm_hall, ohm_thermal;
-Field3D gamma_par_i, heatf_par_i,
-    heatf_par_e; // particle flux, ion and elelctron heat flux
-
-BoutReal hyperresist;  // Hyper-resistivity coefficient (in core only)
-BoutReal ehyperviscos; // electron Hyper-viscosity coefficient
-Field3D hyper_eta_x;   // Radial resistivity profile
-Field3D hyper_eta_z;   // Toroidal resistivity profile
-
-int damp_width;        // Width of inner damped region
-BoutReal damp_t_const; // Timescale of damping
-
-// Metric coefficients
-Field2D Rxy, Bpxy, Btxy, B0, hthe;
-Field2D I;         // Shear factor
-BoutReal LnLambda; // ln(Lambda)
-// Field3D LnLambda;
-
-const BoutReal PI = 3.14159265;
-const BoutReal MU0 = 4.0e-7 * PI;
-BoutReal Mi = 1.6726e-27;        // Ion mass
-const BoutReal KB = 1.38065e-23; // Boltamann constant
-const BoutReal ee = 1.602e-19;   // ln(Lambda)
-const BoutReal eV_K = 11605.0;   // 1eV = 11605K
-
-// Communication objects
-FieldGroup comms;
-
-/// Solver for inverting Laplacian
-std::unique_ptr<Laplacian> phiSolver{nullptr};
-std::unique_ptr<Laplacian> aparSolver{nullptr};
-
-void advect_tracer(const Field3D& p, // phi (input)
-                   const Field3D& delta_x,
-                   const Field3D& delta_z,        // Current location (input)
-                   Field3D& F_dx, Field3D& F_dz); // Time-derivative of location
-
-const Field3D Grad2_par2new(const Field3D& f); // for 4th order diffusion
-
-const Field2D N0tanh(BoutReal n0_height, BoutReal n0_ave, BoutReal n0_width,
-                     BoutReal n0_center, BoutReal n0_bottom_x);
-
-const Field3D field_larger(const Field3D& f, const BoutReal limit);
-
-const Field3D field_larger(const Field3D& f, const BoutReal limit) {
-  Field3D result;
-  result.allocate();
-
-  for (auto i : result) {
-    if (f[i] >= limit)
-      result[i] = f[i];
-    else
-      result[i] = limit;
+#include "bout/constants.hxx"
+#include "bout/physicsmodel.hxx"
+
+#include <cmath>
+
+constexpr BoutReal eV_K = 11605.0; // 1eV = 11605K
+
+class Elm_6f : public PhysicsModel {
```

Comment:
warning: constructor does not initialize these fields: n0_height, n0_ave, n0_width, n0_center, n0_bottom_x, Tconst, laplace_alpha, Tau_ie, q95_input, local_q, q_alpha, n0_fake_prof, T0_fake_prof, Zi, emass, emass_inv, coef_jpar, delta_e, delta_e_inv, gyroAlv, density, Bbar, Lbar, Tbar, Va, Nbar, Tibar, Tebar, dia_fact, diffusion_par, diffusion_perp, diffusion_n4, diffusion_ti4, diffusion_te4, diffusion_v4, diffusion_u4, heating_P, hp_width, hp_length, sink_vp, sp_width, sp_length, sink_Ul, su_widthl, su_lengthl, sink_Ur, su_widthr, su_lengthr, viscos_par, viscos_perp, hyperviscos, Psipara1, Upara0, Upara1, Upara2, Upara3, Nipara1, Tipara1, Tipara2, Tepara1, Tepara2, Tepara3, Tepara4, Vepara, Vipara, Low_limit, gamma_i_BC, gamma_e_BC, Sheath_width, const_cse, include_curvature, include_jpar0, compress0, evolve_pressure, continuity, gyroviscous, diffusion_coef_Hmode0, diffusion_coef_Hmode1, vacuum_pressure, vacuum_trans, nonlinear, evolve_jpar, g, phi_curv, bm_exb, bm_mag, bracket_method_exb, bracket_method_mag, diamag, energy_flux, energy_exch, diamag_phi0, thermal_force, eHall, AA, Vt0, Vp0, experiment_Er, nogradparj, filter_z, filter_z_mode, low_pass_z, zonal_flow, zonal_field, zonal_bkgd, relax_j_vac, relax_j_tconst, smooth_j_x, filter_nl, jpar_bndry_width, parallel_lagrange, parallel_project, phi_constraint, vac_lund, core_lund, vac_resist, core_resist, spitzer_resist, output_transfer, output_ohm, output_flux_par, hyperresist, ehyperviscos, damp_width, damp_t_const, LnLambda [cppcoreguidelines-pro-type-member-init]

```cpp
class Elm_6f : public PhysicsModel {
      ^
```